src/python/zquantum/core/utils.py

```diff
@@ -440,8 +443,26 @@ def create_object(specs: Dict, **kwargs):
     module = importlib.import_module(module_name)
     creator_name = specs.pop("function_name")
     creator = getattr(module, creator_name)
-    created_object = creator(**specs, **kwargs)
-    return created_object
+
+    for key in specs.keys():
+        if key in kwargs.keys():
+            raise ValueError("Cannot have same parameter assigned to multiple values")
+
+    if isinstance(creator, FunctionType):
```

Comment:
Does this check work as expected if `creator` isn't a `def` or `lambda` but an arbitrary object with `__call__` method?

Reply:
I'm actually not sure what the desired behavior would be in this case... 
Playing around with it, it would not classify a class with `__call__` as a function:
```
>>> from types import FunctionType
>>> class MyClass():
...     def __init__(self):
...             pass
...     def __call__(self):
...             pass
... 
>>> isinstance(MyClass(), FunctionType)
False
```

Reply:
Do I understand correctly that the point of this `if` is to be able to call `inspect.signature(creator)` later on on the object? If so, perhaps [EAFP](https://devblogs.microsoft.com/python/idiomatic-python-eafp-versus-lbyl/) with a localized `try`/`catch` with exceptions that can be raised by `inspect.signature` would be more robust against edge cases?

Reply:
One example I can think of where you have a callable class is where you use `AnsatzBasedCostFunction` for `cost_function_specs`. In which case you want `create_object` to instantiate the object, not create a partial function. So I guess this scenario would be handled correctly.